include/oneapi/dpl/experimental/kt/esimd_radix_sort_onesweep.h

```diff
@@ -20,6 +20,79 @@
 
 namespace oneapi::dpl::experimental::esimd::impl
 {
+// float, 32bit
+template <typename T, bool __is_ascending, std::enable_if_t<std::is_same<T, float>::value && sizeof(T) == sizeof(::std::uint32_t), int> = 0>
+constexpr T
+__full_sort_identity()
+{
+    static_assert(std::is_same<T, float>::value, "");
+
+    //                                                                    sign bit
+    //                                                                      |      NaN bit
+    //                                                                      |        |
+    //                                                                      V        V
+    static_assert(::std::numeric_limits<T>::max()    == sycl::bit_cast<T>(0b01111111'01111111'11111111'11111111), "");
+    static_assert(::std::numeric_limits<T>::lowest() == sycl::bit_cast<T>(0b11111111'01111111'11111111'11111111), "");
```

Comment:
Perhaps, these static asserts are not needed here. It may be even dangerous, because as far as I know c++ standard does not state that the representation of a floating point number is consistent with IEEE-754.

Reply:
Here is a note in [cppreference](https://en.cppreference.com/w/c/language/arithmetic_types):
> float - single precision floating-point type. Matches [IEEE-754 binary32 format](https://en.wikipedia.org/wiki/Single-precision_floating-point_format) if supported.

Reply:
The main idea of this static assert to give compile error if we have the different ```float``` presentation in memory.

Reply:
What is the purpose of aborting compilation if the representation is different? As far as I see the new sort identity does not depend on it, it just sets all the bits to 1 (with the exception of descending order with the sign bit set to 0).

Reply:
We discussed it offline and agreed to leave it as is for now